pandas/core/indexing.py

```diff
@@ -1936,42 +1935,31 @@ def _setitem_single_column(self, loc: int, value, plane_indexer):
         """
         pi = plane_indexer
```

Comment:
ill need to look more closely, but based on my experience with #45333 (or more accurately, the attempted simplifications to this method that motivated that PR) I'm very skeptical that this can be meaningfully changed without introducing subtle behavior changes

Reply:
It's certainly possible that I am introducing subtle changes here, as it is a really complicated part of the indexing logic and how it interacts with the internals / caching / etc

Some reasons that I _think_ it might be OK in this case:

- The actual logic (the if/elif/else) to determine whether to do an non-inplace (null or full slice) or inplace setitem is kept intact
- The case of a null/full slice didn't actaully change, as it is still using `self.obj._iset_item` as before. The same for the `elif` case of a same-length array. The main reason for the diff here is that I no longer get the column as `ser`. But for those two cases, the `ser` variable was then always overwritten with the original `value`. So except for potential impact on the cache from no longer creating this intermediate series object, the actual code didn't change (it's basically a removal of the line `ser = self.obj._ixs(loc, axis=1)`, and then replacing `ser` with `value`
- The main change in logic is in the `else` branch. But also here I think it is basically doing the same. Before, it was calling `ser._mgr.setitem((pi,), value)`, and setting those resulting values with `self.obj._iset_item(loc, ser)`. Now this is replaced with the single `self.obj._mgr.column_setitem(loc, pi, value)` call + explicit call to `_clear_item_cache`. But under the hood, the new `column_setitem` is also calling `_mgr.setitem`, and then doing a `_mgr.iset(.., inplace=True)` with the result of that. And the `_iset_item` that was originally used also ends up calling `_mgr.iset(.., inplace=True)` (+ a call to `_clear_item_cache`, which is the reason that I added that here explicitly). 
  So in the end, I think the same internal methods are being called, except that I am now creating an intermediate SingleBlockManager in the internals to perform the setitem, instead of creating an intermediate Series at the indexing.py level. 

But if you can think of certain corner cases that I should check, that would be certainly welcome! The tests are all passing (without having to change them, except for the one case you commented on below), so that is somewhat reassuring, but of course we know well that not all corner cases are covered by our tests ...

The main change is thus no longer creating this intermediate Series object, which _will_ have impact on the DataFrame cache. But I suppose that since this method was calling `_clear_item_cache` (either directly or through `.obj._iset_item`) already, that also shouldn't have an impact.

Reply:
thanks for explaining.  i'll take that closer look tomorrow